docs/migration-guide/w3c-vc-v2.md

```diff
@@ -0,0 +1,565 @@
+| **Status Entry** | `StatusList2021Entry` | `BitstringStatusListEntry` |
+| **Status Subject** | `StatusList2021` | `BitstringStatusList` |
+
+## Migration Steps
+
+### 1. Update Dependencies
+
+First, ensure you're using the latest versions of TrustVC packages that support v2.0:
+
+```bash
+npm install @trustvc/trustvc@latest
+```
+
+### 2. Key Pair Migration
+
+#### v1.1 Key Generation (Legacy)
+```typescript
+import { generateKeyPair, VerificationType } from '@trustvc/w3c-issuer';
+
+// v1.1 - BLS12-381 G2 Key
+const legacyKeyPair = await generateKeyPair({
+  type: VerificationType.Bls12381G2Key2020,
+  seedBase58: 'your-seed-here' // optional
+});
+
+console.log(legacyKeyPair);
+// Output:
+// {
+//   type: "Bls12381G2Key2020",
+//   privateKeyBase58: "...",
+//   publicKeyBase58: "...",
+//   seedBase58: "..."
+// }
+```
+
+#### v2.0 Key Generation (Modern)
+```typescript
+import { generateKeyPair, CryptoSuite } from '@trustvc/w3c-issuer';
+
+// v2.0 - ECDSA-SD-2023 with Multikey
+const modernKeyPair = await generateKeyPair({
+  type: CryptoSuite.EcdsaSd2023 // 'ecdsa-sd-2023'
+});
+
+console.log(modernKeyPair);
+// Output:
+// {
+//   type: "Multikey",
+//   secretKeyMultibase: "z...",
+//   publicKeyMultibase: "z..."
+// }
+```
+
+### 3. Credential Issuance Migration
+
+#### v1.1 Credential Issuance
+```typescript
+import { signW3C } from '@trustvc/trustvc';
+
+const credentialV1 = {
+  '@context': [
+    'https://www.w3.org/2018/credentials/v1',
+    'https://w3id.org/security/bbs/v1'
+  ],
+  type: ['VerifiableCredential'],
+  issuer: 'did:web:example.com',
+  issuanceDate: '2024-01-01T00:00:00Z',
+  expirationDate: '2025-01-01T00:00:00Z',
+  credentialSubject: {
+    id: 'did:example:subject',
+    name: 'John Doe',
+    degree: 'Bachelor of Science'
+  }
+};
+
+const legacyKeyPair = {
+  id: 'did:web:example.com#key-1',
+  controller: 'did:web:example.com',
+  type: 'Bls12381G2Key2020',
+  privateKeyBase58: '...',
+  publicKeyBase58: '...'
+};
+
+const { signed: signedV1 } = await signW3C(credentialV1, legacyKeyPair, 'BbsBlsSignature2020');
+```
+
+#### v2.0 Credential Issuance
+```typescript
+import { signW3C } from '@trustvc/trustvc';
+
+const credentialV2 = {
+  '@context': [
+    'https://www.w3.org/ns/credentials/v2',
+    'https://w3id.org/security/data-integrity/v2'
+  ],
+  type: ['VerifiableCredential'],
+  issuer: 'did:web:example.com',
+  validFrom: '2024-01-01T00:00:00Z',
+  validUntil: '2025-01-01T00:00:00Z',
+  credentialSubject: {
+    id: 'did:example:subject',
+    name: 'John Doe',
+    degree: 'Bachelor of Science'
+  }
+};
+
+const modernKeyPair = {
+  '@context': 'https://w3id.org/security/multikey/v1',
+  id: 'did:web:example.com#multikey-1',
+  controller: 'did:web:example.com',
+  type: 'Multikey',
+  secretKeyMultibase: 'z...',
+  publicKeyMultibase: 'z...'
+};
+
+const { signed: signedV2 } = await signW3C(credentialV2, modernKeyPair, 'ecdsa-sd-2023', {
```

Comment:
can we use CryptoSuite.EcdsaSd2023 for 'ecdsa-sd-2023', so user wont type wrongly.